experiments/tribler/channel_download.py

```diff
@@ -0,0 +1,400 @@
+        self.downloaded_torrent = {}
+
+        self.num_peers = -1
+
+        self.id_experiment = os.environ['EXPERIMENT_NAME'].replace(" ", "")
+
+    def start_session(self):
+        super(ChannelDownloadClient, self).start_session()
+        self.session_deferred.addCallback(self.__config_dispersy)
+        self.session_deferred.addErrback(self._logger.error)
+
+        self.num_peers = os.environ.get('das4_instances_to_run', -1)
+        if self.num_peers == -1:
+            self._logger.error("Cannot get var:das4_instances_to_run")
+            self.num_peers = os.environ.get('sync_subscribers_amount', -1)
+            if self.num_peers == -1:
+                self._logger.error("Cannot get var:sync_subscribers_amount")
+                self.num_peers = len(self.get_peers()) or 200
+
+    def __config_dispersy(self, session):
+
+        count_out = 0
+        while self._dispersy is None and count_out < 100:
+            time.sleep(1.0)
+            count_out += 1
+
+        self.session.lm.dispersy = self._dispersy
+        # self.session.lm.init()
+        self.session.get_dispersy = True
+
+        self._dispersy.define_auto_load(ChannelCommunity, self._my_member, (), {"tribler_session": self.session})
+        self._dispersy.define_auto_load(PreviewChannelCommunity, self._my_member, (), {"tribler_session": self.session})
+
+        from Tribler.Core.TFTP.handler import TftpHandler
+
+        self.session.lm.tftp_handler = TftpHandler(self.session, self.endpoint,
+                                                   "fffffffd".decode('hex'), block_size=1024)
+        self.session.lm.tftp_handler.initialize()
+
+        self.session.lm.rtorrent_handler = RemoteTorrentHandler(self.session)
+        self.session.get_dispersy_instance = lambda: self.session.lm.dispersy
+        self.session.lm.rtorrent_handler.initialize()
+
+        self._logger.error("Dispersy configured")
+
+    def start_dispersy(self):
+        DispersyExperimentScriptClient.start_dispersy(self)
+
+    def setup_session_config(self):
+        config = super(ChannelDownloadClient, self).setup_session_config()
+        config.set_state_dir(os.path.abspath(os.path.join(posix.environ.get('OUTPUT_DIR', None) or BASE_DIR, "Tribler-%d") % os.getpid()))
+        config.set_megacache(True)
+        config.set_dht_torrent_collecting(True)
+        config.set_torrent_collecting(False)
+        config.set_torrent_store(True)
+        config.set_enable_multichain(False)
+        config.set_tunnel_community_enabled(False)
+        config.set_channel_community_enabled(False)
+        config.set_preview_channel_community_enabled(False)
+
+        config.set_dispersy(False)
+
+        self.upload_dir_path = path.join(config.get_state_dir(), "..", "upload_dir")
+        if not os.path.exists(self.upload_dir_path):
+            try:
+                os.makedirs(self.upload_dir_path)
+            except OSError:
+                # race condition of creating shared directory may happen
```

Comment:
If you want to prevent this, you should make sure that only one process (client) is creating the directory. Or you can Gumby be responsible for the creation of this directory before you start the experiment.